app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -46,7 +47,7 @@ public boolean checkAndStartUnsentFormsTask(SyncCapableCommCareActivity activity
                                                 boolean userTriggered) {
         SqlStorage<FormRecord> storage = CommCareApplication.instance().getUserStorage(FormRecord.class);
         FormRecord[] records = StorageUtils.getUnsentRecordsForCurrentApp(storage);
-
+        HiddenPreferences.updateLastUploadAttemptTime();
```

Comment:
I'm not sure we want this invocation:
Does this (or the one below) get triggered with **automatic** sends (the ones that happen in the background), or only on **blocking** sends (the ones that happen when the user presses 'sync', and/or the app triggers autosync)?

We don't want any attempt to send forms to invalidate the forced sync attempt, since the app tries to send forms every time the user completes one. We only want users to get 'credit' for a sync attempt with form submissions if they saw the blocking Modal Dialog come up, rather than having those submissions run in the background.

Reply:
This is the common path for both cases you described though this function gets passed an argument `syncAfterwards` which is always false for non-blocking sync. I am now using that to distinguish between the two cases